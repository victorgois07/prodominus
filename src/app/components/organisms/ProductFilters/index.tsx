import { useProducts } from "@/business/hooks/useProducts";
import { useProductFilters } from "@/business/store/useProductFilters";
import { Input } from "../../atoms";
import { CategoryFilter, SortFilter } from "../../molecules";

interface ProductFiltersProps {
  categories?: string[];
  onFilterChange?: (filters: { search: string; category: string }) => void;
}

export function ProductFilters({
  categories: propCategories,
  onFilterChange,
}: ProductFiltersProps = {}) {
  const {
    categories: hookCategories,
    category,
    setCategory,
    sort,
    setSort,
  } = useProducts();

  const { search, setSearch } = useProductFilters();

  const categories = propCategories || hookCategories;
  const handleCategoryChange = (newCategory: string) => {
    setCategory(newCategory);
    onFilterChange?.({ search, category: newCategory });
  };

  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const newSearch = e.target.value;
    setSearch(newSearch);
    onFilterChange?.({ search: newSearch, category: category || "" });
  };

  return (
    <div className="flex flex-col md:flex-row gap-4 mb-6">
      <Input
        type="text"
        placeholder="Search products"
        onChange={handleSearchChange}
        value={search}
        className="mb-6 appearance-none bg-transparent border-none w-full text-gray-700 mr-3 py-1 px-2 leading-tight focus:outline-none"
      />
      <CategoryFilter
        categories={categories}
        selected={category}
        onSelect={handleCategoryChange}
      />
      <SortFilter sort={sort} onSort={setSort} />
    </div>
  );
}
