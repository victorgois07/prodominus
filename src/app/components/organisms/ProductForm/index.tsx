"use client";

import {
  ProductFormData,
  productFormSchema,
} from "@/business/dtos/product.dto";
import { zodResolver } from "@hookform/resolvers/zod";
import Image from "next/image";
import { useEffect, useState } from "react";
import { Controller, useForm } from "react-hook-form";
import { NumericFormat } from "react-number-format";
import { Input } from "../../atoms";
import { FormField } from "../../molecules";

interface ProductFormProps {
  initialValues?: Partial<ProductFormData>;
  onSubmit: (data: ProductFormData) => void;
  loading?: boolean;
  submitLabel?: string;
}

export function ProductForm({
  initialValues = {},
  onSubmit,
  loading,
  submitLabel = "Salvar",
}: ProductFormProps) {
  const [imagePreview, setImagePreview] = useState<string | undefined>(
    initialValues.image
  );
  const [imageError, setImageError] = useState<string | undefined>(undefined);

  const {
    register,
    handleSubmit,
    formState: { errors: formErrors, isDirty, isValid },
    control,
  } = useForm<ProductFormData>({
    resolver: zodResolver(productFormSchema),
    defaultValues: initialValues as ProductFormData,
    mode: "onChange",
  });

  useEffect(() => {
    if (initialValues.image) {
      validateImage(initialValues.image);
    }
  }, [initialValues.image]);

  const validateImage = (url: string) => {
    setImageError(undefined);
    if (!url) {
      setImagePreview(undefined);
      return;
    }
    const img = new window.Image();
    img.onload = () => {
      setImagePreview(url);
      setImageError(undefined);
    };
    img.onerror = () => {
      setImageError("Não foi possível carregar a imagem");
      setImagePreview(undefined);
    };
    img.src = url;
  };

  const onSubmitForm = (data: ProductFormData) => {
    onSubmit(data);
  };

  return (
    <form
      onSubmit={handleSubmit(onSubmitForm)}
      className="space-y-4 max-w-md mx-auto"
    >
      <FormField
        label="Título"
        htmlFor="title"
        error={formErrors.title?.message}
      >
        <Input
          id="title"
          {...register("title")}
          disabled={loading}
          placeholder="Digite o título do produto"
          aria-invalid={!!formErrors.title}
        />
      </FormField>
      <FormField
        label="Preço"
        htmlFor="price"
        error={formErrors.price?.message}
      >
        <Controller
          name="price"
          control={control}
          render={({ field }) => (
            <NumericFormat
              id="price"
              customInput={Input}
              thousandSeparator="."
              decimalSeparator=","
              prefix="R$ "
              decimalScale={2}
              fixedDecimalScale
              allowNegative={false}
              value={field.value}
              onValueChange={(values) => {
                field.onChange(values.floatValue || 0);
              }}
              disabled={loading}
              placeholder="R$ 0,00"
              aria-invalid={!!formErrors.price}
            />
          )}
        />
      </FormField>
      <FormField
        label="Descrição"
        htmlFor="description"
        error={formErrors.description?.message}
      >
        <textarea
          id="description"
          className="w-full border rounded p-2 min-h-[100px] resize-y"
          {...register("description")}
          disabled={loading}
          placeholder="Digite a descrição do produto"
          aria-invalid={!!formErrors.description}
        />
      </FormField>
      <FormField
        label="Categoria"
        htmlFor="category"
        error={formErrors.category?.message}
      >
        <Input
          id="category"
          {...register("category")}
          disabled={loading}
          placeholder="Digite a categoria do produto"
          aria-invalid={!!formErrors.category}
        />
      </FormField>
      <FormField
        label="URL da Imagem"
        htmlFor="image"
        error={formErrors.image?.message || imageError}
      >
        <div className="space-y-2">
          <Input
            id="image"
            {...register("image", {
              onChange: (e) => {
                const url = e.target.value;
                if (url) {
                  validateImage(url);
                } else {
                  setImagePreview(undefined);
                  setImageError(undefined);
                }
              },
            })}
            disabled={loading}
            placeholder="https://exemplo.com/imagem.jpg"
            aria-invalid={!!formErrors.image || !!imageError}
          />
          {imagePreview && (
            <div className="mt-2">
              <Image
                src={imagePreview}
                alt="Preview"
                width={300}
                height={160}
                className="max-w-full h-40 object-contain rounded border"
                onError={() => {
                  setImageError("Erro ao carregar a imagem");
                  setImagePreview(undefined);
                }}
              />
            </div>
          )}
        </div>
      </FormField>
      <button
        type="submit"
        className="w-full bg-green-600 text-white py-2 rounded font-semibold disabled:opacity-50 hover:bg-green-700 transition-colors"
        disabled={loading || !isDirty || !isValid}
      >
        {loading ? "Salvando..." : submitLabel}
      </button>
    </form>
  );
}
