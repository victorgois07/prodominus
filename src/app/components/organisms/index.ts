export * from "./FavoritesManagement";
export * from "./ProductCard";
export * from "./ProductFilters";
export * from "./ProductForm";
export * from "./ProductGrid";
export * from "./ProductManagement";
