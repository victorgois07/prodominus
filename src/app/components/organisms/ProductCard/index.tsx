import { Button } from "@/app/components/atoms";
import { useFavorites } from "@/business/store/useFavorites";
import { Product } from "@/domain/entities/product.entity";
import Image from "next/image";

interface ProductCardProps {
  product: Product;
  onEdit?: (product: Product) => void;
  onDelete?: (product: Product) => void;
  onClick?: (product: Product) => void;
}

export function ProductCard({
  product,
  onEdit,
  onDelete,
  onClick,
}: ProductCardProps) {
  const { isFavorite, addFavorite, removeFavorite } = useFavorites();
  const favorite = isFavorite(product.id);

  return (
    <div
      role="article"
      className="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      onClick={() => onClick?.(product)}
    >
      <div className="absolute top-2 left-2 z-10">
        <Button
          type="button"
          className={`p-1 rounded-full ${
            favorite ? "bg-red-100 text-red-600" : "bg-gray-100 text-gray-400"
          }`}
          onClick={() =>
            favorite ? removeFavorite(product.id) : addFavorite(product)
          }
          aria-label={
            favorite ? "Remover dos favoritos" : "Adicionar aos favoritos"
          }
        >
          {favorite ? "‚ù§Ô∏è" : "ü§ç"}
        </Button>
      </div>
      <div className="absolute top-2 right-2 z-10 flex gap-2">
        {onEdit && (
          <Button
            className="bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
            onClick={() => onEdit(product)}
            title="Editar"
            type="button"
            aria-label="edit"
          >
            ‚úé
          </Button>
        )}
        {onDelete && (
          <Button
            className="bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
            onClick={() => onDelete(product)}
            title="Excluir"
            type="button"
            aria-label="delete"
          >
            üóëÔ∏è
          </Button>
        )}
      </div>
      <div className="relative h-48 w-full">
        <Image
          src={product.image || "/images/placeholder.svg"}
          alt={product.title}
          fill
          className="object-contain p-4"
        />
      </div>
      <div className="p-4">
        <h3 className="text-lg font-semibold text-gray-800 line-clamp-2 mb-2">
          {product.title}
        </h3>
        <p className="text-gray-600 text-sm line-clamp-2 mb-4">
          {product.description}
        </p>
        <div className="flex justify-between items-center">
          <span className="text-xl font-bold text-green-600">
            R$
            {product.price.toLocaleString("pt-BR", {
              minimumFractionDigits: 2,
              maximumFractionDigits: 2,
            })}
          </span>
          <span className="text-sm text-gray-500 capitalize">
            {product.category}
          </span>
        </div>
        {product.rating && (
          <div className="mt-2 flex items-center text-sm text-gray-500">
            <span className="mr-1">‚òÖ</span>
            <span>{product.rating.rate}</span>
            <span className="mx-1">‚Ä¢</span>
            <span>{product.rating.count} reviews</span>
          </div>
        )}
      </div>
    </div>
  );
}
