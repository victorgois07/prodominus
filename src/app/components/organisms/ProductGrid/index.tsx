import { Product } from "@/domain/entities/product.entity";
import { ProductCard } from "../ProductCard";

interface ProductGridProps {
  products: Product[];
  onEdit?: (product: Product) => void;
  onDelete?: (product: Product) => void;
}

export function ProductGrid({ products, onEdit, onDelete }: ProductGridProps) {
  if (products.length === 0) {
    return (
      <div className="text-center text-gray-500 py-8">No products found</div>
    );
  }

  return (
    <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-6">
      {products.map((product) => (
        <ProductCard
          key={product.id}
          product={product}
          onEdit={onEdit}
          onDelete={onDelete}
        />
      ))}
    </div>
  );
}
