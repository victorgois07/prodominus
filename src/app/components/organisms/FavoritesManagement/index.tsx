import { useFavorites } from "@/business/store/useFavorites";
import { useState } from "react";
import { Modal } from "../../molecules";
import { ProductGrid } from "../ProductGrid";

export function FavoritesManagement() {
  const { favorites } = useFavorites();
  const [isFavoritesOpen, setFavoritesOpen] = useState(false);

  return (
    <>
      <button
        onClick={() => setFavoritesOpen(true)}
        className="bg-pink-600 text-white px-4 py-2 rounded font-semibold hover:bg-pink-700 transition"
      >
        Favoritos ({favorites.length})
      </button>

      <Modal open={isFavoritesOpen} onClose={() => setFavoritesOpen(false)}>
        <h2 className="text-xl font-bold mb-4">Favoritos</h2>
        {favorites.length === 0 ? (
          <p className="text-gray-500">Nenhum produto favoritado.</p>
        ) : (
          <ProductGrid products={favorites} />
        )}
      </Modal>
    </>
  );
}
