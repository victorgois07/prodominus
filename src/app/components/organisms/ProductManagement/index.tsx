import { ProductFormData } from "@/business/dtos/product.dto";
import { useProducts } from "@/business/hooks/useProducts";
import { useConfirmDialog } from "@/business/store/useConfirmDialog";
import { Product } from "@/domain/entities/product.entity";
import { useCallback, useState } from "react";
import { Toast } from "../../atoms";
import { DeleteAllButton } from "../../atoms/DeleteAllButton";
import { Modal } from "../../molecules";
import { FavoritesManagement } from "../FavoritesManagement";
import { ProductForm } from "../ProductForm";
import { ProductGrid } from "../ProductGrid";

type ToastType = {
  message: string;
  type: "success" | "error";
} | null;

export function ProductManagement() {
  const {
    products,
    createProduct,
    updateProduct,
    deleteProduct,
    formLoading,
    formError,
  } = useProducts();
  const confirmDialog = useConfirmDialog();

  const [isModalOpen, setModalOpen] = useState(false);
  const [productBeingEdited, setProductBeingEdited] = useState<Product | null>(
    null
  );
  const [toast, setToast] = useState<ToastType>(null);

  const showToast = useCallback(
    (message: string, type: "success" | "error") => {
      setToast({ message, type });
    },
    []
  );

  const handleEdit = useCallback((product: Product) => {
    setProductBeingEdited(product);
    setModalOpen(true);
  }, []);

  const handleDelete = useCallback(
    async (product: Product) => {
      confirmDialog.show(
        `Deseja realmente excluir o produto "${product.title}"?`,
        async () => {
          try {
            await deleteProduct(product.id);
            showToast("Produto excluído com sucesso!", "success");
          } catch {
            showToast("Erro ao excluir produto.", "error");
          }
        }
      );
    },
    [deleteProduct, showToast, confirmDialog]
  );

  const handleSubmit = useCallback(
    async (data: ProductFormData) => {
      try {
        if (productBeingEdited) {
          await updateProduct(productBeingEdited.id, data);
          showToast("Produto editado com sucesso!", "success");
        } else {
          await createProduct(data);
          showToast("Produto criado com sucesso!", "success");
        }
        setModalOpen(false);
        setProductBeingEdited(null);
      } catch {
        showToast("Erro ao salvar produto.", "error");
      }
    },
    [productBeingEdited, createProduct, updateProduct, showToast]
  );

  const handleModalClose = () => {
    setModalOpen(false);
    setProductBeingEdited(null);
  };

  return (
    <>
      <div className="mb-6 flex justify-between items-center">
        <div className="flex gap-4">
          <button
            onClick={() => setModalOpen(true)}
            className="bg-green-600 text-white px-4 py-2 rounded font-semibold hover:bg-green-700 transition"
          >
            Novo Produto
          </button>
          <DeleteAllButton />
        </div>
        <FavoritesManagement />
      </div>

      <ProductGrid
        products={products}
        onEdit={handleEdit}
        onDelete={handleDelete}
      />

      <Modal open={isModalOpen} onClose={handleModalClose}>
        <h2 className="text-xl font-bold mb-4">
          {productBeingEdited ? "Editar Produto" : "Novo Produto"}
        </h2>
        <ProductForm
          initialValues={productBeingEdited ?? {}}
          onSubmit={handleSubmit}
          loading={formLoading}
          submitLabel={productBeingEdited ? "Salvar alterações" : "Criar"}
        />
        {formError && <div className="text-red-500 mt-2">{formError}</div>}
      </Modal>

      {toast && (
        <Toast
          message={toast.message}
          type={toast.type}
          onClose={() => setToast(null)}
        />
      )}
    </>
  );
}
