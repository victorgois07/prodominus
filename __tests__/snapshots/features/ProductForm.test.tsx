import { ProductForm } from "@/app/components/organisms/ProductForm";
import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

describe("ProductForm Snapshot", () => {
  const mockInitialValues = {
    title: "Test Product",
    price: 99.99,
    description: "Test Description",
    category: "test",
    image: "test.jpg",
  };

  const mockOnSubmit = vi.fn();

  it("should match snapshot with initial values", () => {
    const { container } = render(
      <ProductForm initialValues={mockInitialValues} onSubmit={mockOnSubmit} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with empty initial values", () => {
    const { container } = render(
      <ProductForm
        initialValues={{
          title: "",
          price: 0,
          description: "",
          category: "",
          image: "",
        }}
        onSubmit={mockOnSubmit}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with loading state", () => {
    const { container } = render(
      <ProductForm
        initialValues={mockInitialValues}
        onSubmit={mockOnSubmit}
        loading={true}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with validation errors", () => {
    const { container } = render(
      <ProductForm
        initialValues={mockInitialValues}
        onSubmit={mockOnSubmit}
        errors={{
          title: "Title is required",
          price: "Price must be positive",
          description: "Description is required",
          category: "Category is required",
          image: "Invalid image URL",
        }}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with long values", () => {
    const longValues = {
      title:
        "This is a very long product title that should be properly displayed in the form",
      price: 999999.99,
      description:
        "This is a very long product description that should be properly displayed in the form. It contains multiple sentences and should be properly formatted.",
      category:
        "This is a very long category name that should be properly displayed in the form",
      image:
        "https://example.com/very/long/image/path/that/should/be/properly/displayed/in/the/form.jpg",
    };

    const { container } = render(
      <ProductForm initialValues={longValues} onSubmit={mockOnSubmit} />
    );

    expect(container).toMatchSnapshot();
  });
});
