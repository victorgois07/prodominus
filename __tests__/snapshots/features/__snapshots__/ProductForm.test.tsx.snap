// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductForm Snapshot > should match snapshot with empty initial values 1`] = `
<div>
  <form
    class="space-y-4 max-w-md mx-auto"
  >
    <div>
      <label
        class="block mb-1 font-medium"
        for="title"
      >
        Título
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="title"
        name="title"
        placeholder="Digite o título do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="price"
      >
        Preço
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="price"
        inputmode="numeric"
        placeholder="R$ 0,00"
        type="text"
        value="R$ 0,00"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="description"
      >
        Descrição
      </label>
      <textarea
        aria-invalid="false"
        class="w-full border rounded p-2 min-h-[100px] resize-y"
        id="description"
        name="description"
        placeholder="Digite a descrição do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="category"
      >
        Categoria
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="category"
        name="category"
        placeholder="Digite a categoria do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="image"
      >
        URL da Imagem
      </label>
      <div
        class="space-y-2"
      >
        <input
          aria-invalid="false"
          class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
          id="image"
          name="image"
          placeholder="https://exemplo.com/imagem.jpg"
        />
      </div>
    </div>
    <button
      class="w-full bg-green-600 text-white py-2 rounded font-semibold disabled:opacity-50 hover:bg-green-700 transition-colors"
      disabled=""
      type="submit"
    >
      Salvar
    </button>
  </form>
</div>
`;

exports[`ProductForm Snapshot > should match snapshot with initial values 1`] = `
<div>
  <form
    class="space-y-4 max-w-md mx-auto"
  >
    <div>
      <label
        class="block mb-1 font-medium"
        for="title"
      >
        Título
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="title"
        name="title"
        placeholder="Digite o título do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="price"
      >
        Preço
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="price"
        inputmode="numeric"
        placeholder="R$ 0,00"
        type="text"
        value="R$ 99,99"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="description"
      >
        Descrição
      </label>
      <textarea
        aria-invalid="false"
        class="w-full border rounded p-2 min-h-[100px] resize-y"
        id="description"
        name="description"
        placeholder="Digite a descrição do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="category"
      >
        Categoria
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="category"
        name="category"
        placeholder="Digite a categoria do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="image"
      >
        URL da Imagem
      </label>
      <div
        class="space-y-2"
      >
        <input
          aria-invalid="false"
          class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
          id="image"
          name="image"
          placeholder="https://exemplo.com/imagem.jpg"
        />
        <div
          class="mt-2"
        >
          <img
            alt="Preview"
            class="max-w-full h-40 object-contain rounded border"
            height="160"
            src="test.jpg"
            width="300"
          />
        </div>
      </div>
    </div>
    <button
      class="w-full bg-green-600 text-white py-2 rounded font-semibold disabled:opacity-50 hover:bg-green-700 transition-colors"
      disabled=""
      type="submit"
    >
      Salvar
    </button>
  </form>
</div>
`;

exports[`ProductForm Snapshot > should match snapshot with loading state 1`] = `
<div>
  <form
    class="space-y-4 max-w-md mx-auto"
  >
    <div>
      <label
        class="block mb-1 font-medium"
        for="title"
      >
        Título
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        disabled=""
        id="title"
        name="title"
        placeholder="Digite o título do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="price"
      >
        Preço
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        disabled=""
        id="price"
        inputmode="numeric"
        placeholder="R$ 0,00"
        type="text"
        value="R$ 99,99"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="description"
      >
        Descrição
      </label>
      <textarea
        aria-invalid="false"
        class="w-full border rounded p-2 min-h-[100px] resize-y"
        disabled=""
        id="description"
        name="description"
        placeholder="Digite a descrição do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="category"
      >
        Categoria
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        disabled=""
        id="category"
        name="category"
        placeholder="Digite a categoria do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="image"
      >
        URL da Imagem
      </label>
      <div
        class="space-y-2"
      >
        <input
          aria-invalid="false"
          class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
          disabled=""
          id="image"
          name="image"
          placeholder="https://exemplo.com/imagem.jpg"
        />
        <div
          class="mt-2"
        >
          <img
            alt="Preview"
            class="max-w-full h-40 object-contain rounded border"
            height="160"
            src="test.jpg"
            width="300"
          />
        </div>
      </div>
    </div>
    <button
      class="w-full bg-green-600 text-white py-2 rounded font-semibold disabled:opacity-50 hover:bg-green-700 transition-colors"
      disabled=""
      type="submit"
    >
      Salvando...
    </button>
  </form>
</div>
`;

exports[`ProductForm Snapshot > should match snapshot with long values 1`] = `
<div>
  <form
    class="space-y-4 max-w-md mx-auto"
  >
    <div>
      <label
        class="block mb-1 font-medium"
        for="title"
      >
        Título
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="title"
        name="title"
        placeholder="Digite o título do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="price"
      >
        Preço
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="price"
        inputmode="numeric"
        placeholder="R$ 0,00"
        type="text"
        value="R$ 999.999,99"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="description"
      >
        Descrição
      </label>
      <textarea
        aria-invalid="false"
        class="w-full border rounded p-2 min-h-[100px] resize-y"
        id="description"
        name="description"
        placeholder="Digite a descrição do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="category"
      >
        Categoria
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="category"
        name="category"
        placeholder="Digite a categoria do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="image"
      >
        URL da Imagem
      </label>
      <div
        class="space-y-2"
      >
        <input
          aria-invalid="false"
          class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
          id="image"
          name="image"
          placeholder="https://exemplo.com/imagem.jpg"
        />
        <div
          class="mt-2"
        >
          <img
            alt="Preview"
            class="max-w-full h-40 object-contain rounded border"
            height="160"
            src="https://example.com/very/long/image/path/that/should/be/properly/displayed/in/the/form.jpg"
            width="300"
          />
        </div>
      </div>
    </div>
    <button
      class="w-full bg-green-600 text-white py-2 rounded font-semibold disabled:opacity-50 hover:bg-green-700 transition-colors"
      disabled=""
      type="submit"
    >
      Salvar
    </button>
  </form>
</div>
`;

exports[`ProductForm Snapshot > should match snapshot with validation errors 1`] = `
<div>
  <form
    class="space-y-4 max-w-md mx-auto"
  >
    <div>
      <label
        class="block mb-1 font-medium"
        for="title"
      >
        Título
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="title"
        name="title"
        placeholder="Digite o título do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="price"
      >
        Preço
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="price"
        inputmode="numeric"
        placeholder="R$ 0,00"
        type="text"
        value="R$ 99,99"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="description"
      >
        Descrição
      </label>
      <textarea
        aria-invalid="false"
        class="w-full border rounded p-2 min-h-[100px] resize-y"
        id="description"
        name="description"
        placeholder="Digite a descrição do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="category"
      >
        Categoria
      </label>
      <input
        aria-invalid="false"
        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="category"
        name="category"
        placeholder="Digite a categoria do produto"
      />
    </div>
    <div>
      <label
        class="block mb-1 font-medium"
        for="image"
      >
        URL da Imagem
      </label>
      <div
        class="space-y-2"
      >
        <input
          aria-invalid="false"
          class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
          id="image"
          name="image"
          placeholder="https://exemplo.com/imagem.jpg"
        />
        <div
          class="mt-2"
        >
          <img
            alt="Preview"
            class="max-w-full h-40 object-contain rounded border"
            height="160"
            src="test.jpg"
            width="300"
          />
        </div>
      </div>
    </div>
    <button
      class="w-full bg-green-600 text-white py-2 rounded font-semibold disabled:opacity-50 hover:bg-green-700 transition-colors"
      disabled=""
      type="submit"
    >
      Salvar
    </button>
  </form>
</div>
`;
